lib/rules/radix.js

```diff
@@ -6,67 +6,159 @@
+    return radix.type === "Literal" && radix.value === 10;
+}
 
-            if (!(node.callee.name === "parseInt" || (
-                    node.callee.type === "MemberExpression" &&
-                    node.callee.object.name === "Number" &&
-                    node.callee.property.name === "parseInt"
-                )
-            )) {
-                return;
-            }
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    var mode = context.options[0] || MODE_ALWAYS;
+
+    /**
+     * Checks the arguments of a given CallExpression node and reports it if it
+     * offends this rule.
+     *
+     * @param {ASTNode} node - A CallExpression node to check.
+     * @returns {void}
+     */
+    function checkArguments(node) {
+        var args = node.arguments;
 
-            if (node.arguments.length === 0) {
+        switch (args.length) {
+            case 0:
                 context.report({
                     node: node,
                     message: "Missing parameters."
                 });
-            } else if (node.arguments.length < 2 && mode === MODE_ALWAYS) {
-                context.report({
-                    node: node,
-                    message: "Missing radix parameter."
-                });
-            } else if (node.arguments.length > 1 && mode === MODE_AS_NEEDED &&
-                (node.arguments[1] && node.arguments[1].type === "Literal" &&
-                    node.arguments[1].value === 10)
-            ) {
-                context.report({
-                    node: node,
-                    message: "Redundant radix parameter."
-                });
-            } else {
+                break;
 
-                radix = node.arguments[1];
+            case 1:
+                if (mode === MODE_ALWAYS) {
+                    context.report({
+                        node: node,
+                        message: "Missing radix parameter."
+                    });
+                }
+                break;
 
-                // don't allow non-numeric literals or undefined
-                if (radix &&
-                    ((radix.type === "Literal" && typeof radix.value !== "number") ||
-                    (radix.type === "Identifier" && radix.name === "undefined"))
-                ) {
+            default:
+                if (mode === MODE_AS_NEEDED && isDefaultRadix(args[1])) {
+                    context.report({
+                        node: node,
+                        message: "Redundant radix parameter."
+                    });
+                } else if (!isValidRadix(args[1])) {
                     context.report({
                         node: node,
                         message: "Invalid radix parameter."
                     });
                 }
+                break;
+        }
+    }
+
+    return {
```

Comment:
`variable.references` includes only references of `parseInt`. If we don't use `parseInt`, `variable.references` is empty. So this might be more efficient than checking every `CallExpression` :)